atlp-rwanda/hackers-ec-be
# #187419064 A seller should be able to CRUD products

**What does this PR do?**

As a Seller, I should be able to CRUD items for sale, such that the stock can be seen by any requesting client.

**Description of Task to be completed?**

- A seller should be able to add product
- A every should be able to read the products
- A seller should be able to update the product
- A seller should be able to delete the product

**Endpoints:** 

_POST /api/v1/products ---- (or adding/creating items)_ 
_GET /api/v1/products ----- (for getting all products/items)_ 
_GET /api/v1/products/:id ------ (for getting single product/items)_ 
_PATCH /api/v1/products/:id ----- (for updating a single product/items)_ 
_DELETE /api/v1/products/:id ----- (for removing a product in seller’s collection)_ 

**How should this be manually tested?**

- Clone the repo nd cd into it
- Go to this branch
- Run `npm install`
- Check the `.env.example` and update your` .env` file ( Added environment variables for CLOUDINARY )
- Run `npm run dev` to start the application
- Test every endpoint using postman or Use swagger docs on `/api/v1/docs`

**Pivotracker Id**

#187419064
